modules/core/shared/src/main/scala/gem/util/Section.scala

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.util
+
+import cats.arrow.Category
+import monocle.Iso
+
+/**
+ * A normalizing optic, isomorphic to Iso but with different laws, specifically `get`
+ * need not be injective; i.e., distinct inputs may have the same `get` result, which combined
+ * with a subsequent `reverseGet` yield a normalized form for A.
+ */
+@SuppressWarnings(Array("org.wartremover.warts.Null"))
+final case class Section[A, B](get: A => B, reverseGet: B => A) {
+
+  /** Compose with another Section. */
+  def composeSection[C](f: Section[B, C]): Section[A, C] =
+    Section(a => f.get(get(a)), reverseGet compose f.reverseGet)
```

Comment:
Am I missing something here? Couldn't the new `get` parameter be written as:
`f.get compose get`?

Reply:
(Or maybe this breaks a law due to the fact that the function may not be injective?)

Reply:
Yes, thanks.